services/bundle_analysis/new_notify/contexts/comment.py

```diff
@@ -0,0 +1,156 @@
+        self._notification_context = BundleAnalysisCommentNotificationContext(
+            commit=commit,
+            current_yaml=current_yaml,
+            gh_app_installation_name=gh_app_installation_name,
+        )
+        return self
+
+    async def load_enriched_pull(self) -> "BundleAnalysisCommentContextBuilder":
+        """Loads the EnrichedPull into the NotificationContext.
+        EnrichedPull includes updated info from the git provider and info saved in the database.
+        Raises:
+            NotificationContextBuildError: failed to get EnrichedPull.
+                This can be because there's no Pull saved in the database,
+                or because we couldn't update the pull's info from the git provider.
+        """
+        pull: (
+            EnrichedPull | None
+        ) = await fetch_and_update_pull_request_information_from_commit(
+            self._notification_context.repository_service,
+            self._notification_context.commit,
+            self._notification_context.current_yaml,
+        )
+        if pull is None:
+            raise NotificationContextBuildError("load_enriched_pull")
+        self._notification_context.pull = pull
+        return self
+
+    def load_bundle_comparison(self) -> "BundleAnalysisCommentContextBuilder":
+        """Loads the BundleAnalysisComparison into the NotificationContext.
+        BundleAnalysisComparison is the diff between 2 BundleAnalysisReports,
+        respectively the one for the pull's base and one for the pull's head.
+        Raises:
+            NotificationContextBuildError: missing some information necessary to create
+                the BundleAnalysisComparison.
+        """
+        pull = self._notification_context.pull
+        try:
+            comparison = ComparisonLoader(pull).get_comparison()
+            self._notification_context.bundle_analysis_comparison = comparison
+            return self
+        except (
+            MissingBaseCommit,
+            MissingHeadCommit,
+            MissingBaseReport,
+            MissingHeadReport,
+        ) as exp:
+            raise NotificationContextBuildError(
+                "load_bundle_comparison", detail=exp.__class__.__name__
+            )
+
+    def evaluate_has_enough_changes(self) -> "BundleAnalysisCommentContextBuilder":
+        """Evaluates if the NotificationContext includes enough changes to send the notification.
+        Configuration is done via UserYaml.
+        If a comment was previously made for this PR the required changes are bypassed so that we
+        update the existing comment with the latest information.
+        Raises:
+            NotificationContextBuildError: required changes are not met.
+        """
+        current_yaml = self._notification_context.current_yaml
+        required_changes: bool | Literal["bundle_increase"] = (
+            current_yaml.read_yaml_field(
+                "comment", "require_bundle_changes", _else=False
```

Comment:
Optional: using keyword arguments for what comment and require_bundle_changes here are. While it's a bit more verbose, I find it helps me read the code better, but it's also a stylistic preference so I'm okay either way

Reply:
What do you mean by keyword arguments?
for `"comment" and "require_bundle_changes"` ? You can't do that in this particular function I don't think...

Reply:
Oh I see, read_yaml_field accepts a positional argument in `def read_yaml_field(self, *keys, _else=None)`, so you can't explicitly assign `keys=(...)`, you'd have to modify the fn's internals. Okay it makes sense to leave it as is since the fn expects positional arguments